src/test/scala-2/chiselTests/experimental/InlineTestSpec.scala

```diff
@@ -402,12 +388,111 @@ class InlineTestSpec extends AnyFlatSpec with FileCheck {
       | CHECK:      module RawModuleWithTests
       | CHECK-NEXT:   output io
       |
-      | CHECK:      public module test_RawModuleWithTests_foo
+      | CHECK:      public module test_RawModuleWithTests_passing
       | CHECK-NEXT:   input clock : Clock
       | CHECK-NEXT:   input reset : UInt<1>
       | CHECK-NEXT:   output finish : UInt<1>
       | CHECK-NEXT:   output success : UInt<1>
       """
     )
   }
+
+  def assertPass(result: TestResult.Type): Unit = result match {
+    case other: TestResult.Failure =>
+      fail(s"Test unexpectedly failed: ${other}")
+    case TestResult.Success => ()
+  }
+
+  def assertFail(expectedMessage: String)(result: TestResult.Type): Unit = result match {
+    case TestResult.Success =>
+      fail("Test unexpectedly passed")
+    case TestResult.Failure(actualMessage) if !actualMessage.contains(expectedMessage) =>
+      fail(s"'${actualMessage}' does not match '${expectedMessage}'")
+    case _ => ()
+  }
+
+  it should "simulate and pass if finish asserted with success=1" in {
+    val results = simulateTests(
+      new ModuleWithTests,
+      tests = TestChoice.Name("passing"),
+      timeout = 100
+    )
+    assert(results.size == 1, "Expected exactly one test result")
+    assertPass(results.head.result)
+  }
+
+  it should "simulate and fail if finish asserted with success=0" in {
+    val results = simulateTests(
+      new ModuleWithTests,
+      tests = TestChoice.Name("failing"),
+      timeout = 100
+    )
+    assert(results.size == 1, "Expected exactly one test result")
+    assertFail("test signaled failure")(results.head.result)
```

Comment:
As another alternative, you can combine these into one match:

```suggestion
    results.map(_.result) should contain theSameElementsInOrderAs Seq(TestResult.Failure("test signaled failure"))
```

https://www.scalatest.org/user_guide/using_matchers#workingWithSequences is a great read that gives you a lot of powerful ways to write these things.